### PR TITLE
We should set ID attribute when serialize the response

### PR DESCRIPTION
**Pull request by yankewei**
Thank you for you create this repo I think this repo is the most concise in all SAML libraries written in PHP.
I am considering using this repo in my development, then I found some point that maybe will be convenient for other developers whose using the repo in their SSO development.

### CODE REVIEW

**Comment by yankewei** [comment]
Thank you for you create this repo I think this repo is the most concise in all SAML libraries written in PHP.
I am considering using this repo in my development, then I found some point that maybe will be convenient for other developers whose using the repo in their SSO development.

**Comment by yankewei** [comment]
@william-suppo

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /composer.lock
 /vendor/
 /.phpunit.result.cache
+/.idea
+/.vscode
```
Comment (william-suppo) [inline .gitignore:5]: Plz use global git ignore.

File: composer.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
         "php": ">=7.4",
         "robrichards/xmlseclibs": "~2.0|~3.0|~4.0",
         "symfony/http-foundation": "~5.0|~6.0",
-        "psr/event-dispatcher": "^1.0"
+        "psr/event-dispatcher": "^1.0",
+        "ext-dom": "*"
```
Comment (william-suppo) [inline composer.json:39]: Your PR doesn't talk about ext-dom requirement. Can you remove this line ?

File: src/LightSaml/Model/Protocol/Response.php [hunk 3]
```diff
@@ -129,6 +130,10 @@ public function addEncryptedAssertion(EncryptedElement $encryptedAssertion)
 
     public function serialize(\DOMNode $parent, SerializationContext $context)
     {
+        if ($this->getID() === null) {
+            $this->setID(Helper::generateID());
+        }
```
Comment (william-suppo) [inline src/LightSaml/Model/Protocol/Response.php:135]: Why do you need to have an id in all cases ?

Comment (yankewei) [reply src/LightSaml/Model/Protocol/Response.php:135]: The ID attribute is required for Response and Assertion, Right?

**Pull request closed without merging**